### ISSUE TITLE
No video with vo=gl for a particular file, then segfault upon stepping back in playlist

### ISSUE THREAD

**Issue by commander-derp** [mpv-player/mpv#5534]
### mpv version and platform
mpv-0.27.0 on Linux (Gentoo x86_64)

I tried to build the latest git commit, but it requires newer versions of libav/ffmpeg that make things difficult for me, and I compared to the latest release (tagged 0.28.2, but that appears to be a typo for 0.27.2, according to the changelog) and there are no changes to relevant files, thus I'm sure it's broken on the latest release as well.

### Reproduction steps
I have a particular video file (which, unfortunately, I cannot share at the moment) which renders no video and gives a shader compilation error, and upon playing another file in a playlist (via the '>' key) and returning (with '<') mpv then segfaults.

### Expected behavior
Video plays as normal, and no crashes. Or, if the shader compilation problem is not in mpv, then it should fail gracefully instead of crashing.

### Actual behavior
When using vo=gl, mpv renders no video and gives an error relating to shader compilation. Mplayer and mpv using other video plugins such as vo=x11 work fine.

The relevant lines in the log are:
```
[vo/opengl] fragment shader compile log (status=0):
[vo/opengl] 0:57(7): error: no matching function for call to `mix(float, float, bool)'; candidates are:
[vo/opengl] 0:57(7): error:    float mix(float, float, float)
[vo/opengl] 0:57(7): error:    vec2 mix(vec2, vec2, float)
[vo/opengl] 0:57(7): error:    vec3 mix(vec3, vec3, float)
[vo/opengl] 0:57(7): error:    vec4 mix(vec4, vec4, float)
[vo/opengl] 0:57(7): error:    vec2 mix(vec2, vec2, vec2)
[vo/opengl] 0:57(7): error:    vec3 mix(vec3, vec3, vec3)
[vo/opengl] 0:57(7): error:    vec4 mix(vec4, vec4, vec4)
```

I don't know how OpenGL shader compilation works, but I suspect it may be handled by another library (mesa, perhaps?) and thus the root of the problem may not be in mpv, but even if that is the case, there is still the issue of the segfault that should be fixed so that it would at least fail gracefully and point a finger at the actual problem.

By building with debug symbols and catching the crash in gdb, I think I have found the cause. It appears that at the beginning of gl_renderpass_run() in video/out/opengl/ra_gl.c, the params->pass object is null, so attempting to use pass->priv on the next line triggers the segfault. Since I don't know much about how all this is supposed to work, I decided to stop chasing the rabbit down the hole here. Perhaps a simple null check is needed, or maybe the problem is better resolved further up somewhere.

### Log file

Log file of mpv compiled with debug symbols, and run with -v in gdb, plus a demonstration at the end of what appears to me to be a null pointer causing the crash:
[http://sprunge.us/hRQg](http://sprunge.us/hRQg)

### Sample files
[not currently available]

**Comment by wm4** [comment]
Try 0.28, then git master.

**Comment by commander-derp** [comment]
As I said in the comments for version info, 0.28 appears to be a mis-tagged 0.27 version, and has no relevant changes anyway. Git master causes problems and requires me to upgrade several packages on my system to their git master versions which breaks other things. I guess I could try setting up a new system in a chroot so it doesn't break things...

**Comment by kevmitch** [comment]
> As I said in the comments for version info, 0.28 appears to be a mis-tagged 0.27 version,

I don't think so
[v0.27.2](https://github.com/mpv-player/mpv/commits/release/0.27)
[v0.28.2](https://github.com/mpv-player/mpv/commits/release/0.28)

> and has no relevant changes anyway

How do you know? Did you look at all 579 commits?

>  I guess I could try setting up a new system in a chroot so it doesn't break things...

Try [mpv-build](https://github.com/mpv-player/mpv-build). You don't need to install anything system-wide.

> http://sprunge.us/hRQg

Sorry to ask, given that the issue template explicitly says use sprunge.us, but it has been completely broken of late. Could you upload to https://0x0.st instead.

**Comment by kevmitch** [comment]
https://0x0.st/sT26.txt

**Comment by commander-derp** [comment]
>How do you know? Did you look at all 579 commits?
It seems I was using github's compare function incorrectly, as when I used to compare 0.27 and 0.28, it says there are only 6 commits and shows that 0.28's VERSION file says 0.27.2... however, when viewing the 0.28 branch's VERSION file directly, it correctly states 0.28. 

I had already started getting a chroot setup, and was able to build the latest git commit, and... it seems to be fixed there. So, this is all pretty much moot now.

I guess I should just close this and mark as fixed in latest git? Or would it be worth it to just add an extra sanity check in ra_gl.c just to prevent further crashes in case other bugs come up?
